delphi-frontend/src/main/java/au/com/integradev/delphi/symbol/resolve/NameResolver.java

```diff
@@ -782,9 +782,17 @@ private void handleParenthesizedExpression(ParenthesizedExpressionNode parenthes
     }
   }
 
-  private boolean handleDefaultArrayProperties(ArrayAccessorNode accessor) {
-    if (!declarations.isEmpty()
-        && declarations.stream().allMatch(NameResolver::isDefaultArrayProperty)) {
+  private boolean handleExplicitArrayProperties() {
```

Comment:
This method is quite confusing for a few reasons:

- It does multiple things (determines if the name is an explicit array property, and also adds declarations if the name is a default array property)
- It doesn't handle explicit array properties, it exits early
- Its handling logic is for default array properties

I think it would probably be suitable to inline the early exit for explicit array properties back into `handleDefaultArrayProperties`, rename this routine to `addDefaultArrayPropertyDeclarations` or similar, and call it after the early exit, immediately before `addResolvedDeclaration`. Thoughts?

Reply:
Makes sense to me.